erpnext/accounts/doctype/purchase_invoice/purchase_invoice.py

```diff
@@ -204,7 +204,7 @@ def po_required(self):
 		if frappe.db.get_value("Buying Settings", None, "po_required") == 'Yes':
 			for d in self.get('items'):
 				if not d.purchase_order:
-					throw(_("Purchase Order number required for Item {0}").format(d.item_code))
+					throw(_("As per the Buying Settings, for creating Purchase Invoice, creating Purchase Order and Purchase Receipt is mandatory for Item {0}").format(d.item_code))
```

Comment:
@vishdha,

Need better message it is still confusing,

Also for add the message for Purchase Receipt in `pr_required` method